File: mobile/android/app/src/main/java/com/cdbl/leavemanager/data/model/DelegationModels.kt
package com.cdbl.leavemanager.data.model

data class DelegationsResponse(
    val delegations: List<DelegationEntry> = emptyList()
)

data class DelegationEntry(
    val id: Int,
    val delegatorId: Int,
    val delegatorName: String,
    val delegateId: Int,
    val delegateName: String,
    val startDate: String,
    val endDate: String,
    val reason: String? = null,
    val createdAt: String
)

File: mobile/ios/CDBLLeaveManager/CDBLLeaveManager/Core/Models/DelegationModels.swift
import Foundation
import Combine

struct DelegationsResponse: Decodable {
    let delegations: [DelegationEntry]?
    let items: [DelegationEntry]?

    var allDelegations: [DelegationEntry] {
        delegations ?? items ?? []
    }
}

struct DelegationEntry: Decodable, Identifiable {
    let id: Int
    let delegatorId: Int?
    let delegatorName: String?
    let delegateId: Int?
    let delegateName: String?
    let startDate: String
    let endDate: String
    let reason: String?
    let createdAt: String?
}
